installers/custom_ipxe/main.go

```diff
@@ -3,22 +3,71 @@ package custom_ipxe
 import (
 	"strings"
 
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/boots/installers"
 	"github.com/tinkerbell/boots/ipxe"
 	"github.com/tinkerbell/boots/job"
+	"github.com/tinkerbell/boots/packet"
 )
 
 func init() {
-	job.RegisterSlug("custom_ipxe", bootScript)
+	job.RegisterInstaller("custom_ipxe", ipxeScript)
+	job.RegisterSlug("custom_ipxe", ipxeScript)
 }
 
-func bootScript(j job.Job, s *ipxe.Script) {
+func ipxeScript(j job.Job, s *ipxe.Script) {
+	logger := installers.Logger("custom_ipxe")
+	if j.InstanceID() != "" {
+		logger = logger.With("instance.id", j.InstanceID())
+	}
+
+	var cfg *packet.InstallerData
+	var err error
+
+	if j.OperatingSystem().Installer == "custom_ipxe" {
+		cfg = j.OperatingSystem().InstallerData
+		if cfg == nil {
+			s.Echo("Installer data not provided")
+			s.Shell()
+			logger.Error(err, "empty installer data")
```

Comment:
err is currently nil here, actually this looks like the only use of `err` at all so no need for `var err error` right?

Reply:
good catch, with all the changes this is no longer needed.